src/engine/Map.cpp

```diff
@@ -0,0 +1,530 @@
+}
+
+float Map::GetHeight(const PLVector2 &pos) {
+    MapTile *tile = GetTile(pos);
+    if(tile == nullptr) {
+        return 0;
+    }
+
+    float tile_x = pos.x - std::floor(pos.x);
+    float tile_y = pos.y - std::floor(pos.y);
+
+    float x = tile->height[0] + ((tile->height[1] - tile->height[0]) * tile_x);
+    float y = tile->height[2] + ((tile->height[3] - tile->height[2]) * tile_x);
+    float z = x + ((y - x) * tile_y);
+
+    return z;
+}
+
+void Map::Reset() {
+#if 0 /* todo: redo */
+    SVClearActors();
+
+    for(unsigned int i = 0; i < map_state.num_spawns; ++i) {
+        Actor *actor = Actor_Spawn();
+        if(actor == NULL) {
+            /* warn, and try to keep going for as long as we can :( */
+            LogWarn("failed to spawn actor, probably a memory issue? aborting\n");
+            break;
+        }
+
+        actor->position.x = map_state.spawns[i].position[0];
+        actor->position.y = map_state.spawns[i].position[1];
+        actor->position.z = map_state.spawns[i].position[2];
+
+        actor->angles.x   = map_state.spawns[i].angles[0];
+        actor->angles.y   = map_state.spawns[i].angles[1];
+        actor->angles.z   = map_state.spawns[i].angles[2];
+
+        //actor->bounds   = map_state.spawns[i].bounds;
+        //actor->bounds   = map_state.spawns[i].bounds;
+        //actor->bounds   = map_state.spawns[i].bounds;
+
+        actor->team = map_state.spawns[i].team;
+
+        strncpy(actor->name, map_state.spawns[i].name, sizeof(actor->name));
+        /* todo, setup the actor here - name represents class - this will then
+         * set the model and other parms here
+         */
+
+        //actor->logic.spawn_delay = map_state.spawns[i].spawn_delay;
+    }
+#endif
+}
+
+void Map::LoadSpawns(const std::string &path) {
+    std::ifstream ifs(path, std::ios_base::in | std::ios_base::binary);
+    if(!ifs.is_open()) {
+        Error("Failed to open actor data, \"%s\", aborting!\n", path.c_str());
+    }
+
+    uint16_t num_indices;
+    try {
+        ifs.read(reinterpret_cast<char *>(&num_indices), sizeof(uint16_t));
+    } catch(std::ifstream::failure &err) {
+        Error("Failed to read POG indices count, \"%s\", aborting!\n%s (%d)\n", err.what(), err.code().value());
+    }
+
+    spawns_.reserve(num_indices);
+
+    try {
+        ifs.read(reinterpret_cast<char *>(&spawns_[0]), sizeof(MapSpawn) * num_indices);
+    } catch(std::ifstream::failure &err) {
+        Error("Failed to read POG spawns, \"%s\", aborting!\n%s (%d)\n", err.what(), err.code().value());
+    }
+
+#if 0
+    for(unsigned int i = 0; i < num_indices; ++i) {
+#define v(a) spawns_[i].a[0], spawns_[i].a[1], spawns_[i].a[2]
+        LogDebug("name %s\n", spawns_[i].name);
+        LogDebug("position %d %d %d\n", v(position));
+        LogDebug("bounds %d %d %d\n", v(bounds));
+        LogDebug("angles %d %d %d\n", v(angles));
+        LogDebug("fallback %d %d %d\n", v(fallback_position));
+    }
+#endif
+
+    ifs.close();
```

Comment:
Not needed - file will be closed when ifstream object is destroyed during return